admin_ui/app/org/OrgEditPermissionsModals.js

```diff
@@ -0,0 +1,117 @@
+import React, { useContext, useEffect } from 'react';
+import { useObservable } from 'rxjs-hooks';
+import Store from '../Store';
+import { Link } from '@reach/router';
+import Modal from '../../components/Modal';
+import Input from '../../components/Input';
+import Button from '../../components/Button';
+import Textarea from '../../components/Textarea';
+
+// The CREATE, EDIT or DELETE modals for the OrgEditPermissions page
+const OrgEditPermissionsModals = () => {
+  // Load the store
+  const store = useContext(Store);
+
+  // Establish the value of the currentModal$ observable
+  const currentModal = useObservable(
+    () => store.org.currentPermissionsModal$,
+    store.org.currentPermissionsModal$.getValue()
+  );
+
+  const handleESC = (e) => {
```

Comment:
Shouldn't the handleESC functionality be part of the `components/Modal`? I expect we will need this for all modals going forward. I see you already have an excellent `onClose` prop - why not trigger `onClose` when ESC key is pressed from within the `Modal` component?

Reply:
What a _lovely_ optimisation! Pushed in latest commit.